plugins/destination.go

```diff
@@ -79,55 +75,51 @@ func (p *DestinationPlugin) Init(ctx context.Context, logger zerolog.Logger, spe
 // we implement all DestinationClient functions so we can hook into pre-post behavior
 func (p *DestinationPlugin) Migrate(ctx context.Context, tables schema.Tables) error {
 	SetDestinationManagedCqColumns(tables)
-
-	if p.client == nil {
-		return fmt.Errorf("destination client not initialized")
-	}
-	p.tables = tables
 	return p.client.Migrate(ctx, tables)
 }
 
-func (p *DestinationPlugin) Write(ctx context.Context, sourceName string, syncTime time.Time, res <-chan *schema.Resource) *WriteSummary {
-	if p.client == nil {
-		return nil
+func (p *DestinationPlugin) Read(ctx context.Context, table *schema.Table, sourceName string, res chan<- *schema.DestinationResource) error {
+	SetDestinationManagedCqColumns(schema.Tables{table})
+	return p.client.Read(ctx, table, sourceName, res)
+}
+
+func (p *DestinationPlugin) Write(ctx context.Context, tables schema.Tables, sourceName string, syncTime time.Time, res <-chan *schema.DestinationResource) error {
+	SetDestinationManagedCqColumns(tables)
+	ch := make(chan *schema.DestinationResource)
+	eg := &errgroup.Group{}
+	// given most destination plugins writing in batch we are using a worker pool to write in parallel
+	// it might not generalize well and we might need to move it to each destination plugin implementation.
+	for i := 0; i < writeWorkers; i++ {
```

Comment:
Since `writeWorkers` is always `1` maybe we can remove this for now?

Reply:
yeah we can. I was playing with that, wasn't sure what it should be and how it generalize to most plugins.